proposals/NNNN-HID-Support-Plug-in.md

```diff
@@ -0,0 +1,43 @@
+sharing High level UI/Widget layout information with HU driver with using OnSystemRequest and/or other RPCs (existing RPC). This information may be used for haptic feedback and/or focus management in HU.
+superimposing information on application rendering image. (ex. cursor, focus highlight). These features requirements depend on each specific device features and OEM's UI/UX policy. NOTE) These feature may not be implemented in plugin. NOTE) Some features may be implemented in HU. NOTE) In some case of OEM specific plug-in, existing RPC may be used for not original purpose. However, this is closed spec between specific device and specific plug-in. So, it's not be affect to SDL general spec.
+
+##Potential downsides
+
+Downside:
+OEM should prepare specific plug-in module for their specific device. If not prepared then, downside imapct is only "the special device will not be applicable use. Full functionality of the device cannot be realized". However, this proposal includes by-pass plug-in which pass thru event. So, if all of event can be mapped to existing RPC (OnTouchEvent) then, applications can be run as minimum without specific plug-in. Of course, no special device and only touch-panel I/F then application behavior is not different.
+The applications not to use High-level UI/Widget library but to use application's UI library, cannot take benefit of this architecture. However, OEM can disclose each plug-in spec and such application may be able to customize to use plug-in feature directly.
+If some specific device behavior cannot be mapped existing RPCs then, still new RPC may be required.
+
+##Impact on existing code
+
+Will be modified and new source code is not small to support this architecture. However, impact range is limited in "encoder" part. And, from perspective of Application, this architecture is behind "High-level UI/Widget library". So, it's unnecessary to modify Application logic.
+
+##Alternatives considered
+
+Another approach #1: Add some RPC (ex. OnPointerMove) to support a part of existing specifc device. Demerit of this approach: This solution may not cover future devices. In the case, we should consider next "new RPC" continuously. It may cause "flood of un-organized RPCs".
+Another approach #2: Add general RPC or session service to support general existing and future HID. Not in vehicle but in PC/Set-top-box, already existed standard HID spec. If the spec can be migrate to SDL then, it may be enough solution. However, this requires much modification in not only mobile library but also SDL core. And, "in vehicle" is special environment. So, in the future, something new devices which is not considered in other industry may be created. In that time, HID (and RPC) should be considerd again. And, OEM must always worry how to map new devices to general HID spec when new device comes in.
```

Comment:
Please change the third sentence to "If the spec can be migrated to SDL, then it may be enough of a solution."

Please also clarify this sentence: "So, in the future, something new devices which is not considered in other industry may be created."

Please change "considerd" to "considered".